src/axom/core/Array.hpp

```diff
@@ -73,7 +152,17 @@ bool operator!=(const Array<T>& lhs, const Array<T>& rhs);
  *  is a tuple of 1 or more components, Axom provides the MCArray class.
  *
  *  The Array class mirrors std::vector, with future support for GPUs
- *  in-development.
+ *  in-development.  This class is currently being modified to support
+ *  multidimensional arrays that roughly resemble numpy's ndarray.
+ * 
+ *  \see https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html
+ * 
+ *  In the 1-dimensional case, this class will behave like std::vector.
+ *  In higher dimensions some vector-like functionality will be unavailable,
+ *  e.g., @p push_back , and multidimensional-specific operations will mirror
+ *  ndarray when possible.  In the future, it will be possible to take "views"
+ *  of the underlying array data that allow for flexible reinterpretation via
+ *  different striding.
```

Comment:
Great docs!

Reply:
I'm almost wondering if this (or an even more in-depth conceptual explanation) would belong in the Sphinx docs, so users don't have to dig through the Doxygen for "design philosophy" notes.

Reply:
@joshessman-llnl I think this sort of documentation is best provided in the sphinx user docs for the reson you gave.

Reply:
I agree